### ISSUE TITLE
Connect Bank Account - Endless loading is shown when navigate to Connect Bank Account

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#12670]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue was found when executing PR** https://github.com/Expensify/App/pull/12634

## Action Performed:
Precondition: Add VBA in Web for your testing account
1. Launch the app
2. Log in with your testing account
3. Navigate to Setting - Workspace - Workspaces
4. Tap Connect Bank Account

## Expected Result:
Able to see message "You're all set!" under Connect Bank Account Page

## Actual Result:
Endless loading is shown when navigate to Connect Bank Account

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS
- Android

**Version Number:** 1.2.27.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/201385293-323d7261-3efe-40c0-8f13-d1d7b76891e2.MP4

https://user-images.githubusercontent.com/93399543/201385298-b4f020a5-fec0-4582-897a-da4f0d4ba3ea.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
Checked in IOS production app and works fine, evidence attached 

https://user-images.githubusercontent.com/93399543/201385513-7098ed42-30cb-4df7-acb6-808c0bd41e0f.MP4

**Comment by Luke9389** [comment]
Investigating now.

**Comment by Luke9389** [comment]
The main branch did not have this problem at noon yesterday. Looking at the PRs merged since then....

**Comment by ctkochan22** [comment]
it may also be this PR? https://github.com/Expensify/App/pull/11751

**Comment by ctkochan22** [comment]
@kbecciv Which accounts were you using?

**Comment by Luke9389** [comment]
I'm also unable to reproduce on iOS dev.

**Comment by Luke9389** [comment]
Going to demote this until we can get more reliable reproduction steps.

**Comment by kbecciv** [comment]
I used account: applausetester+1107abb1@applause.expensifail.com.
Reproduction steps:
1. Go to staging.new.expensify.com on Web
2. Go to Settings - Workspace
3. Select Connect Bank account
4. Select Plaid option
5. Select Fidelity option
6. Use credentials: user_good/pass_good
7. Select Saving account -1111
8. Use necessary information for  Alberta Bobbeth Charleson (please ask if you don't have)
9. Finish to sets up the VBA
10. Go to IOS/Andriod app with the same account
11. Go Settings - Workspace -Connect Bank Account

**Comment by kbecciv** [comment]
@ctkochan22 
applausetester+1107abb1@applause.expensifail.com
Password: Feya86Katya

**Comment by Luke9389** [comment]
Thanks for providing more detail on the reproduction steps @kbecciv

**Comment by Luke9389** [comment]
I'm blocked on this because I'm [unable to skip the Onfido step](https://expensify.slack.com/archives/C03TQ48KC/p1668459051592419) on dev (not staging). @roryabraham @yuwenmemon

**Comment by Luke9389** [comment]
Hmm, I'm now experiencing the infinite spinner just after following the SO steps [here](https://stackoverflow.com/c/expensify/questions/342/525#525) (to create an OPEN bank account).

**Comment by Luke9389** [comment]
@kbecciv can you start with a fresh account and see if you get the same thing? 

I did this: 
3. To create an OPEN bank account (this state means the bank account is ready for reimbursement):

- Add Bank Account (On new dot enable Settings > Preferences > Use Secure staging server to hit our staging secure sandbox):

    - Click "Log into your bank"

    - Plaid modal - Choose Wells Fargo (use Bank of America if Wells Fargo is not available)

    - Plaid modal - User Id: user_good

    - Plaid modal - Password: pass_good

    - Plaid modal - Submit

    - Plaid modal - If they request a verification code via mobile sms, enter: credential_good when required

    - Choose an account: Plaid Saving11122XXXXXX111

    - Save & Continue

- Company Information (Enter anything unless specified):

    - Legal Business Name: Alberta Bobbeth Charleson

    - Company website: Enter any valid formatted website

    - Tax ID number: 123456789 (any nine digits)

    - Industry classification code: 123456 (any six digits)

    - Confirm company is not on list of restricted businesses

- Requestor Information (Enter anything unless specified)

    - First name: Alberta

    - Second name: Charleson

    - Toggle confirmation you are authorized to use company bank account

    - Save & Continue (Note: You will receive an error asking to "Please verify you name and date of birth". Hit "Save & Continue" again without changing any inputs.)

- Onfido (skipped)

- Beneficial Owners

    - Toggle terms and conditions

    - Toggle confirmation that the information provided is true and accurate
    
Once the modal closes, you're back at New Dot with an infinite loading spinner. 


If this is happening for you too, then I think we should change this to say it's effecting all platforms (not just iOS).

**Comment by Luke9389** [comment]
Hmm, OK so it looks like the web spinner is happening when you add a VBA that you already have. This screenshot is from an earlier branch that doesn't have the spinner problem (so we can see the error). If you add a VBA successfully (say, on a totally new account, this doesn't happen).
<img width="1306" alt="Screen Shot 2022-11-14 at 2 45 56 PM" src="https://user-images.githubusercontent.com/49007721/201788449-9addb191-b7d6-42b2-8adb-dfc94f56436d.png">

**Comment by Luke9389** [comment]
I'm starting to suspect these might be two different infinite spinners. Testing the original post from above now that I have successfully linked a VBA. (i.e. I'm running iOS to see if the spinner occurs).

**Comment by Luke9389** [comment]
I was able to use different data to get passed the above screen, but now I'm stuck on this one (I've added a validated secondary login, and this still shows up). 

<img width="374" alt="Screen Shot 2022-11-14 at 4 02 15 PM" src="https://user-images.githubusercontent.com/49007721/201793465-ea1a9d5c-745c-440a-9729-be12bc455d44.png">

**Comment by Luke9389** [comment]
When I run iOS, I also get the above screen. So far, I've not been able to reproduce this locally as described in the description. I haven't seen a single infinite spinner just yet.

**Comment by ctkochan22** [comment]
- Logs
    - Not much in the logs -[ Logs of App](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2276a661a73c3e9825-SJC%22)+AND+timestamp:[2022-11-15T06:47:54.205Z+TO+2022-11-15T08:47:54.205Z]&index=logs_expensify-019518) - [Logs of Auth request](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2276a6609ffde29825-SJC%22)+AND+timestamp:[2022-11-15T06:47:12.061Z+TO+2022-11-15T08:47:12.061Z]&index=logs_expensify-019518)
    - ` Free plan verified bank account found -- 1781402` but nothing else to indicate an infinite spinner specific to iOS
- Logging in and testing with applause tester's account.
    - Reproducable **only on iOS**

**Comment by kbecciv** [comment]
@Luke9389 Good day! What would you like me to check?

**Comment by ctkochan22** [comment]
That's so strange that its no longer reproducible. Maybe it was fixed by a recent deploy. However, I've logged into the account and was not able to reproduce.

**Comment by roryabraham** [comment]
Since this is no longer reproducible, I'm going to demote it from deploy blocker to regular bug.

**Comment by ctkochan22** [comment]
@kbecciv let us know if you can reproduce again. But so fart testing on that account we haven't been able to, where as a day ago we were able to. Going to close for now

**Comment by kbecciv** [comment]
@Luke9389 I don't have any loading spinner when added VBA in Web/Chrome.

https://user-images.githubusercontent.com/93399543/202030025-d0d6c7d5-5910-4b58-8a74-ff6887f66d4b.mp4

**Comment by JmillsExpensify** [comment]
Awesome work on this one!

**Comment by kbecciv** [comment]
Able to reproduced the issue in Android/ IOS apps with version 1.2.31.8 


https://user-images.githubusercontent.com/93399543/204031194-d04d79b1-9fba-450e-a1c3-e8fd53439194.MOV

**Comment by JmillsExpensify** [comment]
@ctkochan22 What do you think? Should we create a fresh issue for this one?

**Comment by ctkochan22** [comment]
Yeah lets create a fresh issue.

@kbecciv can you provide your username and password?

**Comment by kbecciv** [comment]
Kbecciv+1031abb2@gmail.com/Feya86Katya

**Comment by JmillsExpensify** [comment]
Alright, new issue created and linked immediately above. I'm closing this issue out.

**Comment by ctkochan22** [comment]
@kbecciv Can you please answer the follwoing and post in the new issue: https://github.com/Expensify/App/issues/13166#issuecomment-1331603051

1. Can you reproduce on other accounts? Or is it specific to this one?
2. What is the username and password of this account?
